### PR TITLE
fix: action run async

### LINKED ISSUE #78: [Question] Is it possible to keep the order of multiple actions in one step ?
### 🧐 问题描述 Problem Description

The `actions` option allows multiple actions separated by comma. I found that the order of execution of the actions is not guaranteed.

However, the order of those actions could be important in some cases. For example, if I want to close and then lock an issue, I might configure like this:

```yaml
      - uses: actions-cool/issues-helper@v2.3.1
        with:
          actions: 'close-issue, lock-issue'
          token: ${{ secrets.GITHUB_TOKEN }}
          issue-number: ${{ github.event.issue.number }}
          body: |
            Hello @${{ github.event.issue.user.login }}. This issue is marked as `invalid` and closed. Please follow the issue template.
```

The issue was firstly locked, and then closed, causing that the bot could not leave a comment to a locked issue:

image

---

So maybe we could:

- Add an extra option like `parallel: false`
- Or, add a suggestion in docs, tell users to use multiple steps in such cases?

### PR DESCRIPTION
**Pull request by xrkffgg**
### 🤔 这个变动的性质是？/ What is the nature of this change?

- [ ] 新特性提交 / New feature
- [x] bug 修复 / Fix bug
- [ ] 样式优化 / Style optimization
- [ ] 代码风格优化 / Code style optimization
- [ ] 性能优化 / Performance optimization
- [ ] 构建优化 / Build optimization
- [ ] 网站、文档、Demo 改进 / Website, documentation, demo improvements
- [ ] 重构代码或样式 / Refactor code or style
- [ ] 测试相关 / Test related
- [ ] 其他 / Other

### 🔗 相关 Issue / Related Issue
fix https://github.com/actions-cool/issues-helper/issues/78

### 💡 需求背景和解决方案 / Background or solution

保持 action 同步执行。

### 📝 更新日志 / Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇨🇳 Chinese |  修复 action 运行异步问题。         |

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -90,13 +90,13 @@ async function main() {
     const actions = core.getInput('actions', { required: true });
 
     const actionsArr = actions.split(',');
-    actionsArr.forEach(item => {
-      testActions(item.trim());
-    });
+    for await (let action of actionsArr) {
```
Comment (meteorlxy) [inline src/main.js]:
```suggestion
    for await (const action of actionsArr) {
```

**Comment by meteorlxy** [comment]
Should we provide an extra option for running in parallel or serially?

**Comment by xrkffgg** [comment]
> Should we provide an extra option for running in parallel or serially?

serially

File: src/main.js [hunk 2]
```diff
@@ -90,13 +90,13 @@ async function main() {
     const actions = core.getInput('actions', { required: true });
 
     const actionsArr = actions.split(',');
-    actionsArr.forEach(item => {
-      testActions(item.trim());
-    });
+    for await (const action of actionsArr) {
+      testActions(action.trim());
+    }
```
Comment (meteorlxy) [inline src/main.js]: I wonder if `for await .. of` loop will take effect here? 🤔

Comment (meteorlxy) [reply src/main.js]: The `for await .. of` loop is used for iterating over async iterable objects. In our case, the "async iterable object" should be `actionsArr`.

However, the `actionsArr` is not a async iterable. It's just a common sync iterable array.

So I think we should simply use `await` in `for .. of` loop:

```ts
for (const action of actionsArr) {
  await testActions(action.trim());
}
```

Comment (xrkffgg) [reply src/main.js]: https://developer.mozilla.org/zh-CN/docs/Web/JavaScript/Reference/Statements/for-await...of

Comment (meteorlxy) [reply src/main.js]:
```js
const arr = [1, 2, 3, 4];

const func = (foo) =>
  new Promise((resolve) => {
    setTimeout(() => {
      console.log(foo);
      resolve();
    }, 2000);
  });

const main = async () => {
  for await (const item of arr) {
    func(item);
  }

  for (const item of arr) {
    await func(item);
  }
};

main();
```

File: src/main.js [hunk 3]
```diff
@@ -90,13 +90,13 @@ async function main() {
...
+      await testActions(action.trim());
+    }
 
-    function testActions(action) {
+    async function testActions(action) {
       if (ALLACTIONS.includes(action)) {
-        choseActions(action);
+        await choseActions(action);
       } else {
         core.setFailed('This actions not supported!');
```
Comment (meteorlxy) [inline src/main.js]:
```suggestion
        core.setFailed(`Action '${action}' is not supported!`);
```

感觉可以顺便改一下 hhh

**Review by meteorlxy — APPROVED** [review]

**Review by Amour1688 — APPROVED** [review]

**Pull request merged**